crates/astria-sequencer/src/authority/state_ext.rs

```diff
@@ -0,0 +1,175 @@
+use std::collections::HashMap;
+
+use anyhow::{
+    anyhow,
+    Context,
+    Result,
+};
+use async_trait::async_trait;
+use borsh::{
+    BorshDeserialize,
+    BorshSerialize,
+};
+use penumbra_storage::{
+    StateRead,
+    StateWrite,
+};
+use proto::native::sequencer::v1alpha1::{
+    Address,
+    ADDRESS_LEN,
+};
+use serde::{
+    Deserialize,
+    Serialize,
+};
+use tracing::instrument;
+
+/// Newtype wrapper to read and write an address from rocksdb.
+#[derive(BorshSerialize, BorshDeserialize, Debug)]
+struct SudoAddress([u8; ADDRESS_LEN]);
+
+/// Newtype wrapper to read and write a validator set or set of updates from rocksdb.
+///
+/// Contains a map of hex-encoded public keys to validator updates.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub(crate) struct ValidatorSet(HashMap<String, tendermint::validator::Update>);
+
+impl ValidatorSet {
+    pub(crate) fn new_from_updates(updates: Vec<tendermint::validator::Update>) -> Self {
+        let mut validator_set = HashMap::new();
+        for update in updates {
+            validator_set.insert(update.pub_key.to_hex(), update);
+        }
+        Self(validator_set)
+    }
+
+    #[allow(dead_code)]
```

Comment:
For private functions it's actually ok to do:
```rust
#[cfg(test)]
pub(crate) fn len(&self) -> usize {
```
It's only problematic in integration tests

Reply:
oh wow, i did not know this, thank you!!